### PR TITLE
feat: add movie and book unit tests

### PR DESCRIPTION
**Pull request by DFraczek**
# Project Updates: Movie and Book Unit Tests

## Overview
This update includes the addition of new endpoints to the `MovieController` and `BookController` for retrieving movies and books by specific criteria. Additionally, unit tests have been implemented for both `BookService` and `MovieService` to ensure correct functionality.

---

## New Endpoints
   - **BookController**:
     - `GET /api/book/{title}` (Get book by Ttitle).

   - **MovieController**:
     - `GET /api/movie/{title}` (Get movie by Ttitle).

### BookService Tests
- **Book Creation Test**: Ensures that a book can be created successfully with a given `BookCreateDto`.
- **Find Book by ID Test**: Validates that a book can be retrieved by a valid ID.
- **Update Book Test**: Verifies that an existing book can be updated with new information.
- **Delete Book Tests**:
  - Throws `EntityNotFoundException` when trying to delete a non-existing book.
  - Successfully deletes an existing book.

### MovieService Tests
- **Movie Creation Test**: Ensures that a movie can be created successfully with a given `MovieCreateDto`.
- **Find Movie by ID Test**: Validates that a movie can be retrieved by a valid ID. Throws `EntityNotFoundException` for an invalid ID.
- **Update Movie Test**: Verifies that an existing movie can be updated with new information.
- **Delete Movie Tests**:
  - Throws `EntityNotFoundException` when trying to delete a non-existing movie.
  - Successfully deletes an existing movie.

## Future Plans
More tests will be added in the future

### CODE REVIEW

**Comment by blackspyek** [comment]
Change commit name to have : instead of /

**Comment by blackspyek** [comment]
Instead of showing whole methods in description, just show endpoints

File: library/src/test/java/org/pollub/library/BookServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        bookCreateDto.setIsbn("123-456-789");
+        bookCreateDto.setDescription("Test Description");
+
+        book = new Book();
+        book.setId(1L);
+        book.setTitle("Test Book");
+    }
+
+    @Test
+    @DisplayName("Ensure correct book creation with given BookCreateDto")
```
Comment (blackspyek) [inline library/src/test/java/org/pollub/library/BookServiceTest.java:51]: So the book in preps is diffrent than a book you are trying to create, so its not ensured that correct book is createad, something is not okay here :C

File: library/src/test/java/org/pollub/library/BookServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        Book foundBook = bookService.findById(1L);
+
+        verify(bookRepository).findById(1L);
+        assertNotNull(foundBook);
+        assertEquals(1L, foundBook.getId());
+        assertEquals("Test Book", foundBook.getTitle());
+    }
+
+    @Test
+    @DisplayName("Successfully update an existing book with given ID and BookCreateDto")
```
Comment (blackspyek) [inline library/src/test/java/org/pollub/library/BookServiceTest.java]: here aswell i don't think this is good test

File: library/src/test/java/org/pollub/library/MovieServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        movieCreateDto.setResolution("1080p");
+        movieCreateDto.setDuration(120);
+        movieCreateDto.setFileFormat("MP4");
+        movieCreateDto.setGenre("Action");
+        movieCreateDto.setShelfNumber(1);
+        movieCreateDto.setDigitalRights("Test DigitalRights");
+    }
+
+    @Test
+    @DisplayName("Ensure correct movie creation")
```
Comment (blackspyek) [inline library/src/test/java/org/pollub/library/MovieServiceTest.java]: same as in bookservice test

File: library/src/test/java/org/pollub/library/MovieServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    @Test
+    @DisplayName("Throw EntityNotFoundException for invalid movie ID")
+    void givenInvalidMovieId_whenFindById_thenThrowEntityNotFoundException() {
+        when(movieRepository.findById(1L)).thenReturn(Optional.empty());
+        assertThrows(EntityNotFoundException.class, () -> movieService.findById(1L));
+    }
+
+    @Test
+    @DisplayName("Update movie successfully with given ID and MovieCreateDto")
```
Comment (blackspyek) [inline library/src/test/java/org/pollub/library/MovieServiceTest.java]: ^^

**Comment by blackspyek** [comment]
Also i would create some unit tests for a edge cases, for example if the Entity is not found instead of throwing 500, i rather throw 404 status code instead

**Review by blackspyek — CHANGES_REQUESTED** [review]
Fix fhings, also check other comments outside review :)

File: library/src/test/java/org/pollub/library/BookServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        Book foundBook = bookService.findById(1L);
+
+        verify(bookRepository).findById(1L);
+        assertNotNull(foundBook);
+        assertEquals(1L, foundBook.getId());
+        assertEquals("Test Book", foundBook.getTitle());
+    }
+
+    @Test
+    @DisplayName("Ensure correct book creation and update with given BookCreateDto")
```
Comment (blackspyek) [inline library/src/test/java/org/pollub/library/BookServiceTest.java]: So its not unit test if you test both methods in one

Comment (blackspyek) [reply library/src/test/java/org/pollub/library/BookServiceTest.java]: just make it only updateTitle, 'cause you created a create test above'

File: library/src/test/java/org/pollub/library/MovieServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    @Test
+    @DisplayName("Throw EntityNotFoundException for invalid movie ID")
+    void givenInvalidMovieId_whenFindById_thenThrowEntityNotFoundException() {
+        when(movieRepository.findById(1L)).thenReturn(Optional.empty());
+        assertThrows(EntityNotFoundException.class, () -> movieService.findById(1L));
+    }
+
+    @Test
+    @DisplayName("Ensure correct movie creation and update with given MovieCreateDto")
+    void givenMovieIdAndDto_whenCreateMovie_thenUpdateTitle_thenReturnUpdatedMovie() {
```
Comment (blackspyek) [inline library/src/test/java/org/pollub/library/MovieServiceTest.java]: same example as in BookServiceTest

**Review by blackspyek — CHANGES_REQUESTED** [review]
fix

**Review by blackspyek — APPROVED** [review]
Optimize imports and merge
